python/ray/tune/trainable.py

```diff
@@ -511,6 +515,18 @@ def logdir(self):
         """
         return os.path.join(self._logdir, "")
 
+    @property
+    def trial_info(self):
+        """TrialInfo object for the corresponding trial of this Trainable.
+
+        This allows you to access fields such as the trial_id and trial name
+        inside the training loop. See the documentation on TrialInfo for more
+        informatino.
```

Comment:
wait why allow the ID and name to change? shouldn't these be read-only

also typo (information)

Reply:
Fixed; changed to be exposed as properties.